### PR TITLE
[Runtime] Pack element count into the flags of swift_getTupleTypeMetadata

### PR DESCRIPTION
**Pull request by DougGregor**
Reduces the # of parameters we need to pass to this runtime API.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

File: include/swift/ABI/MetadataValues.h [hunk 1]
```diff
@@ -690,20 +690,29 @@ using ParameterFlags = TargetParameterTypeFlags<uint32_t>;
 template <typename int_type>
 class TargetTupleTypeFlags {
   enum : int_type {
-    NonConstantLabelsMask    = 1 << 0,
+    NumElementsMask = 0x0000FFFFU,
+    NonConstantLabelsMask = 0x01000000U,
```
Comment (rjmccall) [inline include/swift/ABI/MetadataValues.h]: Why is this bit 24 instead of bit 16?

File: include/swift/Runtime/RuntimeFunctions.def [hunk 2]
```diff
@@ -873,15 +873,14 @@ FUNCTION(GetObjCClassFromMetadata, swift_getObjCClassFromMetadata, DefaultCC,
...
+// Metadata *swift_getTupleTypeMetadata(TupleTypeFlags flags,
 //                                      Metadata * const *elts,
 //                                      const char *labels,
-//                                      uint32_t flags,
 //                                      value_witness_table_t *proposed);
 FUNCTION(GetTupleMetadata, swift_getTupleTypeMetadata, DefaultCC,
          RETURNS(TypeMetadataPtrTy),
-         ARGS(SizeTy, TypeMetadataPtrTy->getPointerTo(0),
-              Int8PtrTy, Int32Ty, WitnessTablePtrTy),
+         ARGS(Int32Ty, TypeMetadataPtrTy->getPointerTo(0),
```
Comment (rjmccall) [inline include/swift/Runtime/RuntimeFunctions.def]: The reason that these functions are generally declared as taking a size_t/uintptr_t is to eliminate potential ABI issues around extending the smaller type to fit the full register.  Please imitate that here (and above in the declaration of the TupleTypeFlags typedef).

File: include/swift/ABI/MetadataValues.h (hunk 1 above)
Comment (DougGregor) [reply include/swift/ABI/MetadataValues.h]: No reason; I'll move it down to 16.

File: include/swift/Runtime/RuntimeFunctions.def (hunk 2 above)
Comment (DougGregor) [reply include/swift/Runtime/RuntimeFunctions.def]: Okay!

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Pull request merged**